src/main/java/com/amazon/ion/impl/IonCursorBinary.java

```diff
@@ -1841,13 +1841,19 @@ private boolean checkContainerEnd() {
      * Resets state specific to the current value.
      */
     private void reset() {
+        valueTid = null;
         valueMarker.typeId = null;
         valueMarker.startIndex = -1;
         valueMarker.endIndex = -1;
         fieldSid = -1;
+        fieldTextMarker.typeId = null;
+        fieldTextMarker.startIndex = -1;
+        fieldTextMarker.endIndex = -1;
         hasAnnotations = false;
+        annotationSequenceMarker.typeId = null;
         annotationSequenceMarker.startIndex = -1;
         annotationSequenceMarker.endIndex = -1;
+        annotationTokenMarkers.clear();
```

Comment:
I'd like us to be as surgical as possible here, as I've noticed that the size of this method correlates with overall performance. I'd particularly like to know whether clearing `annotationTokenMarkers` is actually necessary. We can use ion-java-benchmark-cli to see if this change has performance impact.

Reply:
I checked this out locally and ran some tests, and found that just removing `annotationTokenMarkers.clear()` improves performance by up to 5%. I think we can remove this, as we already have the `hasAnnotations` boolean being reset in here, and that boolean gates all public annotation-getter APIs. Additionally, `annotationTokenMarkers` is cleared before new annotations are read in `uncheckedReadAnnotationWrapperHeader_1_1` and `slowReadAnnotationWrapperHeader_1_1`.